modelstore/storage/blob_storage.py

```diff
@@ -243,5 +265,69 @@ def unset_model_state(self, domain: str, model_id: str, state_name: str):
         model_state_path = self._get_metadata_path(domain, model_id, state_name)
         if self._remove(model_state_path):
             logger.debug(
-                "Successfully unset %s=%s from %s", domain, model_id, state_name
+                "Successfully unset %s=%s from state=%s", domain, model_id, state_name
+            )
+        else:
+            logger.debug(
+                "Model  %s=%s was not set to state=%s", domain, model_id, state_name
             )
+
+    def _get_metadata_path(
+        self, domain: str, model_id: str, state_name: Optional[str] = None
+    ) -> str:
+        """Creates a path where a meta-data file about a model is stored.
+        I.e.: :code:`operatorai-model-store/<domain>/versions/<model-id>.json`
+
+        Args:
+            domain (str): A group of models that are trained for the
+            same end-use are given the same domain.
+
+            model_id (str): A UUID4 string that identifies this specific
+            model.
+        """
+        return os.path.join(
+            get_models_path(self.root_prefix, domain, state_name), f"{model_id}.json"
+        )
+
+    def set_meta_data(self, domain: str, model_id: str, meta_data: dict):
+        logger.debug("Setting meta-data for %s=%s", domain, model_id)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            local_path = os.path.join(tmp_dir, f"{model_id}.json")
+            remote_path = self._get_metadata_path(domain, model_id)
+            with open(local_path, "w") as out:
+                out.write(json.dumps(meta_data))
+            self._push(local_path, remote_path)
+
+            # @TODO this is setting the "latest" model implicitly
+            remote_path = get_domain_path(self.root_prefix, domain)
+            self._push(local_path, remote_path)
+
+    def _pull_and_load(self, remote_path: str) -> dict:
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            local_path = self._pull(remote_path, tmp_dir)
+            with open(local_path, "r") as lines:
+                return json.loads(lines.read())
+
+    def get_meta_data(self, domain: str, model_id: str) -> dict:
+        if any(x in [None, ""] for x in [domain, model_id]):
+            raise ValueError("domain and model_id must be set")
+        logger.debug("Retrieving meta-data for %s=%s", domain, model_id)
+        remote_path = self._get_metadata_path(domain, model_id)
+        try:
+            return self._pull_and_load(remote_path)
+        except FilePullFailedException:
+            logger.debug("Failed to pull: %s", remote_path)
+            # A meta-data file may not be downloaded if:
+            # 1. The domain does not exist (e.g., typo)
+            # 2. The model never existed in that domain
+            # 3. The model has been deleted from that domain
+            # 4. A different error occurred (e.g. connectivity)
+            # The block below currently checks for (2) and (3)
```

Comment:
There are two cases that I've left out here - (1) should return a `DomainNotFoundException`, and (4) should raise whatever exception is behind it. Leaving those for the future too.